assessment.sql

```diff
@@ -6,23 +6,46 @@ CREATE DATABASE "sql-crud-assessment";
 -- Question 1. Create a table called `licensees` to hold data in columns surname,given_name,gender,height,weight,dob,eye_color,hair_color.  See `data/licensees.csv` for example data.  Don't forget that all tables should have an id column as a unique identifier for a row.
 
 -- your code here
+CREATE TABLE licensees(
+id SERIAL PRIMARY KEY,
+given_name CHARACTER VARYING,
+gender CHARACTER VARYING (1),
+height INTEGER,
+weight INTEGER
+dob DATE,
+eye_color CHARACTER VARYING,
+hair_color CHARACTER VARYING,
+);
 
 -- Question 2. Add `Bishop, Caren, f, 63, 132, 1943-09-26, Brown, Black` into the table of licensees.
 
 -- your code here
+INSERT INTO licensees(given_name, gender, height, weight, dob, eye_color, hair_color)
+VALUES('Bishop, Caren', 'f', 63, 132, 1943-09-26, 'Brown', 'Black');
 
 -- Question 3. Load the data from `data/licensees.csv` into the `licensees` table.
 
 -- your code here
 
+
 -- Question 4. Write a query to get all attributes of licensees with Hazel eye color and Brown or Black hair color.
 
 -- your code here
+SELECT *
+FROM licensees
+WHERE eye_color = 'Hazel'
+AND hair_color = 'Black' OR 'Brown';
```

Comment:
So, this question asked for a complex condition in the WHERE clause. Here's a breakdown...

Logical operators work on booleans. They're not like the English words; `OR 'Brown'` here is a syntax error because a logical operator is preceded by a boolean (`hair_color = 'Black'`, which evaluates to `TRUE` or `FALSE`) and followed by a string. To make this work, you want to change this clause to `hair_color = 'Black' OR hair_color = 'Brown'`.

Next, operator precedence. After the above correction, this clause works incorrectly. It would actually search for rows where both `eye_color` is `'Hazel`' and `hair_color` is `'Black'`, or just `hair_color` is `'Brown'`. This is because `AND` has higher priority than `OR`. We need to use parens to override priority. So, finally, our revised `WHERE` clause is:

```sql
...
WHERE eye_color = 'Hazel' AND (hair_color = 'Black' OR hair_color = 'Brown')
...
```
3/5